tools/integration_tests/rapid_appends/reads_after_appends_test.go

```diff
@@ -0,0 +1,115 @@
+	"github.com/googlecloudplatform/gcsfuse/v3/tools/integration_tests/util/setup"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (t *CommonAppendsSuite) TestAppendsAndReads() {
+	const metadataCacheTTLSecs = 10
+	metadataCacheEnableFlag := fmt.Sprintf("%s%v", "--metadata-cache-ttl-secs=", metadataCacheTTLSecs)
+	fileCacheDirFlag := func() string {
+		return "--cache-dir=" + getNewEmptyCacheDir(t.primaryMount.rootDir)
+	}
+
+	testCases := []struct {
+		name string
+	}{
+		{
+			name: "SequentialRead",
+		},
+	}
+
+	for _, scenario := range []struct {
+		enableMetadataCache bool
+		enableFileCache     bool
+		flags               []string
+	}{{
+		// all cache disabled
+		enableMetadataCache: false,
+		enableFileCache:     false,
+		flags:               []string{"--write-experimental-enable-rapid-appends=true", "--write-global-max-blocks=-1", "--metadata-cache-ttl-secs=0"},
+	}, {
+		enableMetadataCache: true,
+		enableFileCache:     false,
+		flags:               []string{"--write-experimental-enable-rapid-appends=true", "--write-global-max-blocks=-1", metadataCacheEnableFlag},
+	}, {
+		enableMetadataCache: true,
+		enableFileCache:     true,
+		flags:               []string{"--write-experimental-enable-rapid-appends=true", "--write-global-max-blocks=-1", metadataCacheEnableFlag, "--file-cache-max-size-mb=-1", fileCacheDirFlag()},
+	}, {
+		enableMetadataCache: false,
+		enableFileCache:     true,
+		flags:               []string{"--write-experimental-enable-rapid-appends=true", "--write-global-max-blocks=-1", "--metadata-cache-ttl-secs=0", "--file-cache-max-size-mb=-1", fileCacheDirFlag()},
+	}} {
+		func() {
+			t.mountPrimaryMount(scenario.flags)
+			defer t.unmountPrimaryMount()
+
+			log.Printf("Running tests with flags: %v", scenario.flags)
+
+			for _, tc := range testCases {
+				t.Run(tc.name, func() {
+					// Initially create an unfinalized object.
+					t.createUnfinalizedObject()
+					defer t.deleteUnfinalizedObject()
+
+					// Open this object as a file for appending on the appropriate mount.
+					appendFileHandle := operations.OpenFileInMode(t.T(), path.Join(t.appendMountPath, t.fileName), os.O_APPEND|os.O_WRONLY|syscall.O_DIRECT)
+					defer operations.CloseFileShouldNotThrowError(t.T(), appendFileHandle)
+
+					readPath := path.Join(t.primaryMount.testDirPath, t.fileName)
+					for i := range numAppends {
+						sizeBeforeAppend := len(t.fileContent)
+						t.appendToFile(appendFileHandle, setup.GenerateRandomString(appendSize))
+						sizeAfterAppend := len(t.fileContent)
+
+						gotContent, err := operations.ReadFile(readPath)
+
+						require.NoError(t.T(), err)
+						readContent := string(gotContent)
+						if !scenario.enableMetadataCache || !t.isSyncNeededAfterAppend || (i == 0) {
```

Comment:
can you add comments here on why are you validating full content in few cases and partially in other cases.